libs/ledger-live-common/src/hw/signMessage/index.ts

```diff
@@ -36,10 +36,17 @@ const signMessage: SignMessage = (transport, account, opts) => {
   const { currency } = account;
   let signMessage = perFamily[currency.family].signMessage;
   if ("type" in opts) {
-    signMessage =
-      opts.type === AcreMessageType.Withdraw
-        ? ACREMessageSigner.signWithdraw
-        : ACREMessageSigner.signMessage;
+    switch (opts.type) {
+      case AcreMessageType.Withdraw:
+        signMessage = ACREMessageSigner.signWithdraw;
+        break;
+      case AcreMessageType.SignIn:
+        signMessage = ACREMessageSigner.signIn;
+        break;
+      default:
+        signMessage = ACREMessageSigner.signMessage;
+        break;
+    }
```

Comment:
Shouldn't this logic be downstram in `perFamily[currency.family].signMessage` ? As it's called with the opts it could totally do that.
It's very weird to have coin specific logic in here.

Reply:
We don't have a different family for ACRE unfortunately
ACRE should just be Bitcoin but they forked the app and need to use other code sometimes until we support plugins in a similar way to the Ethereum app
I know all of this is ugly and unwanted but we don't really have an alternative at the moment but if you see one, I'll take it

Reply:
Maybe I can see if we can move this logic as you said